rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -268,6 +272,10 @@ data LanguageService
     | references    (set[loc] (Focus _focus) referencesService)
     | implementation(set[loc] (Focus _focus) implementationService)
     | codeAction    (list[CodeAction] (Focus _focus) codeActionService)
+    | prepareRename (loc (Focus _focus) prepareRenameService)
```

Comment:
How to return a failure here (for unsupported renames)?
- `Maybe[loc]`
- throw an exception
- ...

Reply:
I think throwing an exception is the best option? and otherwise a custom ast:

```rascal
data RenameOption = fullTarget(loc l) | notPossible(Message m);
```

Reply:
I agree. Just throwing a string (failure message) here might be the easiest in most cases?